modules/default/weather/providers/openmeteo.js

```diff
@@ -0,0 +1,541 @@
+					return;
+				}
+
+				const dailyForecast = this.generateWeatherObjectsFromForecast(parsedData);
+				this.setWeatherForecast(dailyForecast);
+			})
+			.catch(function (request) {
+				Log.error("Could not load data ... ", request);
+			})
+			.finally(() => this.updateAvailable());
+	},
+
+	fetchWeatherHourly() {
+		this.fetchData(this.getUrl())
+			.then((data) => this.parseWeatherApiResponse(data))
+			.then((parsedData) => {
+				if (!parsedData) {
+					// No usable data?
+					return;
+				}
+
+				const hourlyForecast = this.generateWeatherObjectsFromHourly(parsedData);
+				this.setWeatherHourly(hourlyForecast);
+			})
+			.catch(function (request) {
+				Log.error("Could not load data ... ", request);
+			})
+			.finally(() => this.updateAvailable());
+	},
+
+	/**
+	 * Overrides method for setting config to check if endpoint is correct for hourly
+	 *
+	 * @param {object} config The configuration object
+	 */
+	setConfig(config) {
+		this.config = {
+			lang: config.lang ?? "en",
+			...this.defaults,
+			...config
+		};
+
+		this.config.ignoreToday = (this.config.ignoreToday ?? false) && ["daily", "forecast"].includes(this.config.type);
+
+		// Set properly maxNumberOfDays and max Entries properties according to config and value ranges allowed in the documentation
+		const maxEntriesLimit = ["daily", "forecast"].includes(this.config.type) ? 7 : this.config.type === "hourly" ? 48 : 0;
+		if (this.config.hasOwnProperty("maxNumberOfDays") && !isNaN(parseFloat(this.config.maxNumberOfDays))) {
+			const daysFactor = ["daily", "forecast"].includes(this.config.type) ? 1 : this.config.type === "hourly" ? 24 : 0;
+			this.config.maxEntries = Math.max(1, Math.min(Math.round(parseFloat(this.config.maxNumberOfDays)) * daysFactor, maxEntriesLimit));
+			this.config.maxNumberOfDays = Math.ceil(this.config.maxEntries / Math.max(1, daysFactor));
+		}
+		this.config.maxEntries = Math.max(1, Math.min(this.config.maxEntries, maxEntriesLimit));
+
+		if (!this.config.type) {
+			Log.error("type not configured and could not resolve it");
+		}
+
+		this.fetchLocation();
+	},
+
+	// Generate valid query params to perform the request
+	getQueryParameters() {
+		let params = {
+			latitude: this.config.lat,
+			longitude: this.config.lon,
+			timeformat: "unixtime",
+			timezone: "auto",
+			past_days: this.config.past_days ?? 0,
+			daily: this.dailyParams,
+			hourly: this.hourlyParams,
+			// Fixed units as metric
+			temperature_unit: "celsius",
+			windspeed_unit: "kmh",
+			precipitation_unit: "mm"
+		};
+
+		const startDateOffset = this.config.ignoreToday ? 1 : 0;
+		const today = moment().startOf("day");
+		const startDate = moment(today).add(startDateOffset, "days");
+
+		switch (this.config.type) {
+			case "hourly":
+				params["start_date"] = startDate.format("YYYY-MM-DD");
+				params["end_date"] = moment(startDate).add(this.config.maxNumberOfDays, "days").format("YYYY-MM-DD");
+				break;
+			case "daily":
+			case "forecast":
```

Comment:
I am trying this in my config:
```
		{
			module: "weather",
			position: "top_right",
			config: {
				appendLocationNameToHeader: false,
				weatherProvider: 'openmeteo',
				ignoreToday: true,
				type: "forecast",
				lat: 52.520008,
				lon: 13.404954,
			}
		},
```
but dont get any data displayed. Using 
```
				type: "forecast",
```
works though, Bug or Feature?

Reply:
A bug. Already fixed on line 303